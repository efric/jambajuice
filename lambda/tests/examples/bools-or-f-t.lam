(\ true false and or not .

    or false true

    -- Should reduce to true, i.e., \ x y . x
)
-- true
(\ x y . x)
-- false
(\ x y . y)
-- and
(\ p q . p q p)
-- or
(\ p q . p p q)
-- not
(\ p x y . p y x)
