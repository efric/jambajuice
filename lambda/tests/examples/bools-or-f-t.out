(\true . \false . \and . \or . \not . or false true) (\x . \y . x) (\x . \y . y) (\p . \q . p q p) (\p . \q . p p q) (\p . \x . \y . p y x)
